File: OP_PROJECT_JAVA/src/game/Animals/Turtle.java
package game.Animals;

import game.Animal;
import game.Organism;
import game.Position;
import game.World;

import java.util.Random;

public class Turtle extends Animal {
    private static final int TURTLE_STRENGTH = 2;
    private static final int TURTLE_INIT = 1;
    private static final String TURTLE_TYPE = "Turtle";


    public Turtle(Position position, World world) {
        super(TURTLE_TYPE, TURTLE_STRENGTH, TURTLE_INIT, position, world);
    }

    @Override
    protected boolean checkReflect(Organism attacker) {
        if ( attacker.getStrength() < 5) {   // Turtle reflects attack
            String info = "Turtle reflected " + attacker.getType();
            world.displayInformation(info);
            return true;
        }
        return false;
    }

    @Override
    public void action() {
        Random random = new Random();
        int chance = random.nextInt(100);
        if (chance <= 75) {
            return; // Turtle stays in the same place (75%)
        }
        super.action(); // Move the turtle
    }
}

File: OP_PROJECT_PYTHON/src/game/Animals/Turtle.py
from random import randint
from src.game.Animal import Animal


class Turtle(Animal):
    TURTLE_STRENGTH = 2
    TURTLE_INIT = 1
    TURTLE_TYPE = "Turtle"

    def __init__(self, position, world):
        super().__init__(Turtle.TURTLE_TYPE, Turtle.TURTLE_STRENGTH, Turtle.TURTLE_INIT, position, world)

    def check_reflect(self, attacker):
        if attacker.get_strength() < 5:
            info = "Turtle reflected " + attacker.get_type()
            self.world.display_information(info)
            return True
        return False

    def action(self):
        chance = randint(0, 99)
        if chance <= 75:
            return
        super().action()
